aws-s3-transfer-manager/src/operation/upload_objects.rs

```diff
@@ -57,7 +57,7 @@ impl UploadObjects {
         };
         let (cancel_tx, cancel_rx) = watch::channel(false);
         let concurrency = handle.num_workers();
-        let ctx = UploadObjectsContext::new(handle.clone(), input, cancel_tx);
+        let ctx = UploadObjectsContext::new(handle.clone(), input, cancel_tx, cancel_rx);
```

Comment:
**nit**: Since you pass both in you could just have `new()` create the watch channel for you.

Reply:
Yeah, I usually prefer dependency injection but in this case, creating receiver ends internally in `UploadObjectsContext` feels the right thing to do (i.e., if a user passed two unrelated receiver ends to `new`, then cancellation mechanism wouldn't work).